pkg/check/node_perf.go

```diff
@@ -215,3 +202,27 @@ func (c *CheckNodeDiskPerf) CheckNode(ctx *CheckContext, node *v1.Node, vm *mo.V
 func (c *CheckNodeDiskPerf) FinishCheck(ctx *CheckContext) {
 	return
 }
+
+// checkNodePerf Check to see if the specified node should have its performance check.
+func checkNodePerf(node *v1.Node, mcd MetricCheckDef) bool {
+	checkPerf := true
+	_, isMaster := node.Labels["node-role.kubernetes.io/master"]
+	_, isWorker := node.Labels["node-role.kubernetes.io/worker"]
+
+	if isMaster && mcd.checkMasters == false {
+		klog.V(2).Infof("skipping metric check of [%s] as it is not intended for master nodes", mcd.metricName)
+		checkPerf = false
+	} else if isWorker && mcd.checkWorkers == false {
+		if isMaster && mcd.checkMasters {
```

Comment:
This check looks useless, `if isMaster && mcd.checkMasters == false` was already checked at line 212 

What about decomposing these nested conditiones into something sane. Did I miss anything?

```
if isMaster && mcd.checkMasters {
  return true
}

if isWorker && mcd.checkWorkers {
  return true
}

return false

Reply:
Line 212 check if `mcd.checkMasters == false`, but line 216 check if `mcd.checkMasters == true`.  From what I can gather, the intent was to have this logic log when its disabled for workers, but enabled for master when a master node is also schedulable with the role=worker.  The logic was moved into this function from the function with the big if loop being fixed in previous command and no logic other than the final else if clause was added.

It did take me several read attempts to make sure I too understood the intention.  I like the logging as a way to help support diagnose when these cases are happening which is what lead to this bug being opened.  No comments documenting the original intent were hear, but I think I am preserving the original master / worker combo with the improvement of not running when its neither a master nor worker.  Let me know your thoughts.

Reply:
Ack, let's keep it the same as the original and anyone can refactor it in a subsequent PR. It just hurts my eyes.